app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+        let show = false;
+
+        if (postIds && postIds.length >= 10) {
+            show = true;
+        }
+
+        if (!show) {
+            this.loadPosts(props, context);
+        }
+
+        this.state = {
+            title: channelName,
+            show,
+            error: '',
+            retry: false
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelName !== nextProps.channelName) {
+            this.setState({title: nextProps.channelName});
+        }
+
+        if (this.props.focusedPostId !== nextProps.focusedPostId) {
+            this.setState({show: false}, () => {
+                if (nextProps.postIds && nextProps.postIds.length < 10) {
+                    this.loadPosts(nextProps);
+                } else {
+                    this.setState({show: true});
+                }
+            });
+        }
+    }
+
+    goToThread = wrapWithPreventDoubleTap((post) => {
+        const {actions, navigator, theme} = this.props;
+        const channelId = post.channel_id;
+        const rootId = (post.root_id || post.id);
+
+        actions.loadThreadIfNecessary(rootId, channelId);
+        actions.selectPost(rootId);
+
+        const options = {
+            screen: 'Thread',
+            animated: true,
+            backButtonTitle: '',
+            navigatorStyle: {
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg
+            },
+            passProps: {
+                channelId,
+                rootId
+            }
+        };
+
+        navigator.push(options);
+    });
+
+    handleClose = () => {
+        const {actions, navigator, onClose} = this.props;
+        if (this.refs.view) {
+            this.refs.view.zoomOut().then(() => {
+                actions.selectPost('');
+                navigator.dismissModal({animationType: 'none'});
+
+                if (onClose) {
+                    onClose();
+                }
+            });
+        }
+    };
+
+    jumpToChannel = (channelId, channelDisplayName) => {
+        if (channelId) {
+            const {actions, channelTeamId, currentTeamId, navigator, onClose, theme} = this.props;
+            const currentChannelId = this.props.channelId;
+            const {
+                handleSelectChannel,
+                handleTeamChange,
+                markChannelAsRead,
+                setChannelLoading,
+                setChannelDisplayName,
+                markChannelAsViewed
+            } = actions;
+
+            if (this.refs.view) {
+                this.refs.view.zoomOut().then(() => {
+                    actions.selectPost('');
+
+                    if (onClose) {
+                        onClose();
+                    }
+
+                    navigator.resetTo({
```

Comment:
I think you could use `navigator.popToRoot` here to skip having to redefine the root screen

Reply:
same as in the user profile